### PR TITLE
[WIP] Suites config flexibility

### PR DESCRIPTION
**Pull request by DavertMik**
This PR allows having a flexible configuration for suites.

#### 1

Suite can be defined inside the main config:

```yaml
actor_suffix: Tester
paths:
    tests: .
    log: _output
    data: _data
    support: _support
suites:
    unit:
        path: .
        actor: UnitTester
        modules:
            enabled:
                - Asserts
```

#### 2

Suite can have its custom path (specified by `path`). From config above expects unit tests to be placed into the root directory, where codeception.yml is located (`path: tests: .` and `path: ``)

selection_201

#### 3

Suite can be defined without an actor, which is useful for unit testing

```yaml
paths:
    tests: tests
    log: tests/_output
    data: tests/_data
    support: tests/_support
    envs: tests/_envs
suites:
    unit:
        path: .
        modules:
            enabled:
                - Asserts
```
In this case UnitTester won't be created, as well as `_generated/UnitActions`. However, such suites won't be able to have Cest and Cept files generated.

#### 4

`class_name` suite in suite config replaced with `actor`:

```
class_name: UnitTester => actor: UnitTester
```

#### 5

In global config `actor` replaced with `actorSuffix` config option (which makes more sense)

### CODE REVIEW

File: src/Codeception/Configuration.php [hunk 1]
```diff
@@ -65,11 +65,12 @@ class Configuration
      * @var array Default config
      */
     public static $defaultConfig = [
-        'actor'      => 'Guy', // codeception 1.x compatibility
+        'actor_suffix'=> 'Guy', // Codeception 1.x compatibility
```
Comment (Naktibalda) [inline src/Codeception/Configuration.php]: Please drop this compatibility with Codeception 1.x

File: src/Codeception/Configuration.php [hunk 2]
```diff
@@ -101,14 +102,15 @@ class Configuration
     ];
 
     public static $defaultSuiteSettings = [
-        'class_name'  => 'NoGuy',
+        'actor'       => null,
+        'class_name'  => null, // Codeception >2.3 compatibility
```
Comment (Naktibalda) [inline src/Codeception/Configuration.php]: Forward compatibility?

File: src/Codeception/Test/Loader/Gherkin.php [hunk 3]
```diff
@@ -66,10 +66,10 @@ protected function fetchGherkinSteps()
             $this->addSteps($roleContexts, "role:$role");
         }
 
-        if (empty($this->steps) and empty($contexts['default'])) { // if no context is set, actor to be a context
+        if (empty($this->steps) and empty($contexts['default']) and $this->settings['actor']) { // if no context is set, actor to be a context
```
Comment (Naktibalda) [inline src/Codeception/Test/Loader/Gherkin.php]: Please replace `and` with `&&`

File: tests/data/bundled_suites/_support/_generated/UnitTesterActions.php [hunk 4]
```diff
@@ -0,0 +1,501 @@
+<?php  //[STAMP] b822c13a1ed14d868ece1edfdbc8bd14
```
Comment (Naktibalda) [inline tests/data/bundled_suites/_support/_generated/UnitTesterActions.php]: Please git ignore generated files and remove this file

**Comment by DavertMik** [comment]
@Naktibalda fixed everything you've mentioned

File: src/Codeception/Configuration.php [hunk 5]
```diff
@@ -181,6 +182,10 @@ public static function config($configFile = null)
 
         self::$config = $config;
 
+        if (isset(self::$config['actor'])) {
+            self::$config['actor_suffix'] = self::$config['actor']; // old compatibility
```
Comment (Naktibalda) [inline src/Codeception/Configuration.php]: Comment could be better

**Comment by DavertMik** [comment]
Goog point!!! :+1:

**Pull request merged**